lighthouse-core/scripts/pptr-run-devtools.js

```diff
@@ -152,122 +176,146 @@ const startLighthouse = `
       }
     }
   }
+}
+
+async function runLighthouse() {
+  // @ts-expect-error global
+  const panel = UI.panels.lighthouse || UI.panels.audits;
+
+  /** @type {Promise<{lhr: LH.Result, artifacts: LH.Artifacts}>} */
+  const resultPromise = new Promise((resolve, reject) => {
+    const methodName = panel.__proto__.buildReportUI ?
+      'buildReportUI' : '_buildReportUI';
+    addSniffer(
+      panel.__proto__,
+      methodName,
+      // @ts-expect-error implicit any
+      (lhr, artifacts) => resolve({lhr, artifacts})
+    );
+
+    addSniffer(
+      panel.statusView.__proto__,
+      'renderBugReport',
```

Comment:
Errors thrown in the Lighthouse bundle will be sent through this function, so we should listen for errors and reject when we find them.